src/core/server/http/http_server.ts

```diff
@@ -339,4 +343,23 @@ export class HttpServer {
       return t.next({ headers: authResponseHeaders });
     });
   }
+
+  private registerStaticDir(path: string, dirPath: string) {
+    if (this.server === undefined) {
+      throw new Error('Http server is not setup up yet');
+    }
+
+    this.server.route({
+      path,
+      method: 'GET',
+      handler: {
+        directory: {
+          path: dirPath,
+          listing: false,
+          lookupCompressed: true,
```

Comment:
Will that be moved under the `ressources` subservice when https://github.com/elastic/kibana/issues/50654 lands?

Reply:
I don't think so. I'd rather keep all `hapi`  specific API in within `http_server`. HttpResource may build its API on top of HttpServer.